edgecast/customer/customer.go

```diff
@@ -0,0 +1,368 @@
+	request, err := svc.Client.BuildRequest(
+		"GET",
+		"v2/pcc/customers/services",
+		nil,
+	)
+
+	if err != nil {
+		return nil, fmt.Errorf("GetAvailableCustomerServices: %v", err)
+	}
+
+	var services []Service
+
+	_, err = svc.Client.SendRequest(request, &services)
+
+	if err != nil {
+		return nil, fmt.Errorf("GetAvailableCustomerServices: %v", err)
+	}
+
+	return &services, nil
+}
+
+// GetCustomerServices gets the list of services available to the provided
+// customer and whether each service is enabled or disabled.
+func (svc *CustomerService) GetCustomerServices(
+	params GetCustomerServicesParams,
+) (*[]Service, error) {
+	relURL := fmt.Sprintf("v2/pcc/customers/%s/services", params.Customer.HexID)
+	request, err := svc.Client.BuildRequest("GET", relURL, nil)
+
+	if err != nil {
+		return nil, fmt.Errorf("GetCustomerServices: %v", err)
+	}
+
+	var services []Service
+
+	_, err = svc.Client.SendRequest(request, &services)
+
+	if err != nil {
+		return nil, fmt.Errorf("GetCustomerServices: %v", err)
+	}
+
+	return &services, nil
+}
+
+// UpdateCustomerServices enables or disables the provided services based on the
+// status provided.
+func (svc *CustomerService) UpdateCustomerServices(
+	params UpdateCustomerServicesParams,
+) error {
+	for _, serviceID := range params.ServiceIDs {
+		relUrl := fmt.Sprintf("v2/pcc/customers/%s/services/%v",
+			params.Customer.HexID, serviceID)
+
+		body := &struct {
+			Status int `json:"Status"`
+		}{
+			Status: params.Status,
+		}
+
+		request, err := svc.Client.BuildRequest("PUT", relUrl, body)
+
+		if err != nil {
+			return fmt.Errorf(
+				"UpdateCustomerServices build request failed. Error: %v\n Body: %v",
+				err, body,
+			)
+		}
+
+		resp, err := svc.Client.SendRequest(request, nil)
+
+		if err == nil && resp.StatusCode != 200 {
+			return fmt.Errorf(
+				"failed to set customer services, please contact an administrator",
+			)
+		}
+
+		if err != nil {
+			return fmt.Errorf(
+				"UpdateCustomerServices send request failed. Error: %v\n Body: %v",
+				err, body)
+		}
+	}
+
+	return nil
+}
+
+// GetCustomerDeliveryRegion retrieves the current active delivery region for
+// the provided customer
+func (svc *CustomerService) GetCustomerDeliveryRegion(
+	params GetCustomerDeliveryRegionParams,
+) (int, error) {
+	relURL := fmt.Sprintf(
+		"v2/pcc/customers/%s/deliveryregions",
+		params.Customer.HexID,
+	)
+
+	request, err := svc.Client.BuildRequest("GET", relURL, nil)
+
+	if err != nil {
+		return 0, fmt.Errorf("GetCustomerDeliveryRegion: %v", err)
+	}
+
+	parsedResponse := &struct {
+		AccountNumber    string `json:"AccountNumber"`
+		CustomID         string `json:"CustomId"`
+		DeliveryRegionID int    `json:"DeliveryRegionId"`
+	}{}
+
+	_, err = svc.Client.SendRequest(request, &parsedResponse)
+
+	if err != nil {
+		return 0, fmt.Errorf("GetCustomerDeliveryRegion: %v", err)
+	}
+
+	return parsedResponse.DeliveryRegionID, nil
```

Comment:
If API returns all three properties, I would like to access all from caller function of GetCustomerDeliveryRegion.
Thinking of readability, if GetCustomerDeliveryRegion() returning just integer value, a new dev has to figure it out what it means. But it returns the response structure, he/she would know that the returning int means DeliveryRegionID w/o searching code base.

Reply:
@changgyu-oh good call, updated.